paramiko/config.py

```diff
@@ -162,6 +163,27 @@ def parse(self, file_obj):
                 # at the end (for compatibility with issue #415). After 3.x, it
                 # will simply not get stripped, leaving a nice explicit marker.
                 context["config"][key] = None
+            elif key == 'include':
+                # Support for Include directive in ssh_config
+                path = value
+                # According to SSH documentation if the path is relative,
+                # we look forward in ~/.ssh
+                if (not os.path.isabs(path)) and path[0] != '~':
+                    path = os.path.join("~", ".ssh", path)
+                # Expand the user home path
+                path = os.path.expanduser(path)
+                if parsed_files is None:
+                    parsed_files = []
+
+                # Parse every included file
+                for filename in glob.iglob(path):
+                    if os.path.isfile(filename):
+                        if filename in parsed_files:
+                            raise ValueError("Include loop detected in ssh "
+                                             "config file: %s" % filename)
+                        with open(filename) as fd:
+                            parsed_files.append(filename)
+                            self.parse(fd, parsed_files)
```

Comment:
i'm trying to wrap my head around the scope of `parsed_files` here and it hurts a little. it *seems* like it's limited to the caller function, so in theory you *could* still create a loop where file A includes file B includes file C includes file A, and *not* be detected... does that make sense?

i wonder if a class variable should be used here instead, in other words. i understand it would break parsing the same file multiple times, but is that even supported?

Reply:
I think this catches a circular inclusion, but a test case would be nice and would make it clear.